### ISSUE TITLE
Fails at stage 3

### ISSUE THREAD

**Issue by StoltzfusJ** [mourisl/Rcorrector#9]
When I run:
`perl run_rcorrector.pl -1 /1RNAseq/read_1.fastq.gz -2 /1RNAseq/read_2.fastq.gz -od /1RNAseq/rCorrector_out/ -t 2 -verbose`
I get the following: 
`Put the kmers into bloom filter
jellyfish bc -m 23 -s 100000000 -C -t 2 -o tmp_d8b9e7e64de6664e34a42a1e89d74e8e.bc <(gzip -cd /1RNAseq/read_1.fastq.gz) <(gzip -cd /1RNAseq/read_2.fastq.gz) 
Count the kmers in the bloom filter
jellyfish count -m 23 -s 100000 -C -t 2 --bc tmp_d8b9e7e64de6664e34a42a1e89d74e8e.bc -o tmp_d8b9e7e64de6664e34a42a1e89d74e8e.mer_counts <(gzip -cd /1RNAseq/read_1.fastq.gz) <(gzip -cd /1RNAseq/read_2.fastq.gz) 
Dump the kmers
jellyfish dump -L 2 tmp_d8b9e7e64de6664e34a42a1e89d74e8e.mer_counts > tmp_d8b9e7e64de6664e34a42a1e89d74e8e.jf_dump
Error correction
/1RNAseq/Programs/Rcorrector-1.0.2/rcorrector -od /1RNAseq/rCorrector_out/ -t 2 -verbose  -p /1RNAseq/read_1.fastq.gz /1RNAseq/read_2.fastq.gz -c tmp_d8b9e7e64de6664e34a42a1e89d74e8e.jf_dump
Failed at stage 3.`
I've tried this with multiple samples and always get a "failed at stage 3" error.  I've run the input reads through FASTQC, and things look good.  Any help would be appreciated.

**Comment by mourisl** [comment]
Do you have the write access to the /1RNAseq/ folder? Since there is no "." in front of it, it points to the root directory which we usually don't have access right.
If this is not the case, can you show me a few lines of the fastq file?

Thanks

**Comment by StoltzfusJ** [comment]
Hi,

Thank you for your reply.  In answer to your questions:

Here are the permissions for the directories:
drwxr-xr-x   4 jonathanstoltzfus root  4096 Jun  9 15:21 1RNAseq
drwxr-xr-x 2 jonathanstoltzfus jonathanstoltzfus        4096 Jun  8 13:23 rCorrector_out

Here are the first few lines of one of the fastq files:
@K00315:112:HTGNJBBXX:6:1101:1357:1595 1:N:0:NTCACG
CNACGATGTTCCGGATGACGAACTTCTTGATGGCCTTGTCCTTGGGGACGCAGCGGGCGCAGTTGGTGCAGCGGATGGGCTGGACGTGCCCGCGGCCCTTCTTAGCGCGCCCGTTGAGATCGGAAGAGCACACGTCCGAACTCCAGTCAC
+
A#AFFJJJFFJJJJJFF<FAJJJJ<FJF<AJF<FJJJFFJJ7<FAAAJJFFFAAFJFJJJJJ--7----FA<--A-7A7<7<<FJ<<-77FFJF-A-A7FAJ7JFJ<FAF-AAAF7<JJFAJ<J-7--7FFFJF-A)7AAF-AFF<-AJF
@K00315:112:HTGNJBBXX:6:1101:1418:1595 1:N:0:NTCACG
CNACACGGATCTCACTCCCAGTGAAACGGCAGGGTGGGAGGATGGGCACCGCATCCTCCTTGTATTTAAAGGGAATTCCACTCATGTGCTGCCTGGCAAGAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATG
+
A#AFFJJJJJJJJJJJJJJJJAJJJJJJFJJJJJ7<JJJJJJJFJJJJFA<J<7FJ-A<--<-7-7<AFJ<A<JJFJFAJJA-A-A<-AFF7JFA<FJJJJ-A7FAF<AFAFJJJA7F<A7<<FF7<AJFFJAFFFAJJJA-77-<<<-A


Thank you for your assistance.

Sincerely,

Jonathan


______________________
Jonathan Stoltzfus, Ph.D.
Assistant Professor of Biology
Millersville University of Pennsylvania
________________________________

**Comment by mourisl** [comment]
Can you show me the first lines of tmp_d8b9e7e64de6664e34a42a1e89d74e8e.jf_dump? And how many lines in that file?

**Comment by StoltzfusJ** [comment]
Hi,

The first 10 lines of tmp_d8b9e7e64de6664e34a42a1e89d74e8e.jf_dump are:
p
>1830207
AAAAAAAAAAAAAAAAAAAAAAA
>3
ACACATCCAGTGTGGTAGATCTA
>5
ACTTGTGCTGGGACACCAAGCGC
>2
GTAGGTTGTGGTGGTATCTCGGA
>3
GCCCCATGTTGCCAACTCCAGAA

The number of lines in the file is 462359212:
T3500:/1RNAseq/Programs/Rcorrector-1.0.2$ wc -l tmp_d8b9e7e64de6664e34a42a1e89d74e8e.jf_dump
462359212 tmp_d8b9e7e64de6664e34a42a1e89d74e8e.jf_dump

Thank you,

Jonathan


________________________________

**Comment by mourisl** [comment]
Can you directly run "/1RNAseq/Programs/Rcorrector-1.0.2/rcorrector -od /1RNAseq/rCorrector_out/ -t 2 -verbose -p /1RNAseq/read_1.fastq.gz /1RNAseq/read_2.fastq.gz -c tmp_d8b9e7e64de6664e34a42a1e89d74e8e.jf_dump" to see whether there is any error message? Sorry for the troubles.

**Comment by StoltzfusJ** [comment]
Hi Li,

I think I figured out the problem.  When I ran “make”, there was an error that I didn’t catch (one of the libraries in the version of Ubuntu that I am running wasn’t updated, resulting in the error).  After updating the library, and re-configuring, I now have rCorrector running.

My apologies for not realizing this earlier.  Thank you for your assistance and your work in making this tool.

Sincerely,

Jonathan


Jonathan Stoltzfus, Ph.D.
Assistant Professor of Biology

MILLERSVILLE UNIVERSITY
P.O. Box 1002, Millersville, PA 17551-0302
Phone: 717-871-4098
https://www.millersville.edu/biology/faculty/j-stoltzfus.php
